src/System.Private.ServiceModel/tools/test/SelfHostWcfService/TestResources/WebSocketResources.cs

```diff
@@ -118,4 +119,88 @@ protected override Binding GetBinding()
+            httpsTransportBindingElement.MaxReceivedMessageSize = DefaultMaxReceivedMessageSize;
+            httpsTransportBindingElement.MaxBufferSize = DefaultMaxReceivedMessageSize;
+            httpsTransportBindingElement.TransferMode = TransferMode.Streamed;
+            CustomBinding binding = new CustomBinding(textMessageEncodingBindingElement, httpsTransportBindingElement);
+            return binding;
+        }
+
+        protected override void ModifyHost(ServiceHost serviceHost, ResourceRequestContext context)
+        {
+            // Ensure the https certificate is installed before this endpoint resource is used
+            CertificateResourceHelpers.EnsureSslPortCertificateInstalled(context.BridgeConfiguration);
+
+            base.ModifyHost(serviceHost, context);
+        }
+    }
+
+    internal class WebSocketHttpRequestReplyTextStreamedResource : EndpointResource<WSRequestReplyService, IWSRequestReplyService>
+    {
+        protected override string Protocol { get { return BaseAddressResource.Http; } }
+
+        protected override int GetPort(ResourceRequestContext context)
+        {
+            return context.BridgeConfiguration.BridgeWebSocketPort;
+        }
+
+        protected override string Address { get { return "WebSocketHttpRequestReplyTextStreamedResource"; } }
+
+        protected override Binding GetBinding()
+        {
+            int DefaultMaxReceivedMessageSize = 64 * 1024 * 1024;
+
+            NetHttpBinding binding = new NetHttpBinding();
+            binding.WebSocketSettings.TransportUsage = WebSocketTransportUsage.Always;
+            binding.MaxReceivedMessageSize = DefaultMaxReceivedMessageSize;
+            binding.MaxBufferSize = DefaultMaxReceivedMessageSize;
+            binding.TransferMode = TransferMode.Streamed;
+            binding.MessageEncoding = NetHttpMessageEncoding.Text;
+            return binding;
+        }
+    }
+
+    internal class WebSocketHttpDuplexTextStreamedResource : EndpointResource<WSDuplexService, IWSDuplexService>
+    {
+        protected override string Protocol { get { return BaseAddressResource.Http; } }
+
+        protected override int GetPort(ResourceRequestContext context)
+        {
+            return context.BridgeConfiguration.BridgeWebSocketPort;
+        }
+
+        protected override string Address { get { return "WebSocketHttpDuplexTextStreamedResource"; } }
+
+        protected override Binding GetBinding()
+        {
+            int DefaultMaxReceivedMessageSize = 64 * 1024 * 1024;
+
+            NetHttpBinding binding = new NetHttpBinding();
+            binding.WebSocketSettings.TransportUsage = WebSocketTransportUsage.Always;
+            binding.MaxReceivedMessageSize = DefaultMaxReceivedMessageSize;
+            binding.MaxBufferSize = DefaultMaxReceivedMessageSize;
+            binding.TransferMode = TransferMode.Streamed;
+            binding.MessageEncoding = NetHttpMessageEncoding.Text;
+            return binding;
```

Comment:
NIT: ditto, consider using the 

```csharp
NetHttpBinding binding = new NetHttpBinding() 
{ 
    MaxReceivedMessageSize = ..., 
    MaxBufferSize = ....
} 
```

syntax so the code looks a little cleaner. Not required.